### ISSUE TITLE
tile cache volume does not work

### ISSUE THREAD

**Issue by cgommel** [Overv/openstreetmap-tile-server#186]
Hello,

everything works fine but one feature does not work: Setting a directory  for the tile cache. 

When starting the docker with
`-v /srv/osm/tiles:/var/lib/mod_tile `

there come errors that the access to the directory is denied. 

But the same syntax:
` -v /srv/osm/data:/var/lib/postgresql/12/main `

works fine for the database folder. I tried on on several machines and it worked nowhere. So there might be a permission issue.

**Comment by bilange** [comment]
I feel like I had a very similar issue as you. But I will comment more broadly to help future users as well. 

TLDR: I was able to resolve the issue for renderd's file permissions this way: 

* Figure out the UID/GID for the renderd process inside the container by running `ps aux` inside the openstreetmap-tile-server running container: 
  * `docker exec -ti osm-tileserver bash`  # note that `osm-tileserver` is the name I decided to give to that container personally, yours WILL be different
  * `ps aux | grep renderd` # note that `ps` works here because this docker container based on ubuntu and contains more command line tools than say alpine or more minimalist docker distros
  * Note that `ps` says that this command is ran by user `renderer` ; typing `cat /etc/passwd | grep renderer` will reveal that renderer has the userid & groupid 1000
* Exit the container, and give user 1000 and group 1000 rights to your tiles volume: `chown -R 1000:1000 /srv/docker/osm-tileserver/tiles` - this path corresponds to whereever you exported your `/var/lib/mod_tile` volume to
* Restart the container, and enjoy!

--

The long story: I followed the documentation and noticed **after its creation** (I already imported my OSM data at that point so postgresql was already populated) that I could export the tiles volume, and created a new volume in my docker config for the tiles. I am under the impression (but I have not verified) that the dockerfile maintainer here probably checks for correct filesystems permission on initial startup and not afterwards.
